vector/src/main/java/im/vector/app/features/roomprofile/RoomProfileViewModel.kt

```diff
@@ -112,6 +117,36 @@ class RoomProfileViewModel @AssistedInject constructor(
                 }
     }
 
+    private fun observeAdminMembers() {
+        val roomMemberQueryParams = roomMemberQueryParams {
+            displayName = QueryStringValue.IsNotEmpty
+            memberships = Membership.activeMemberships()
+        }
+
+        combine(
+                room.flow().liveRoomMembers(roomMemberQueryParams),
+                room.flow()
+                        .liveStateEvent(EventType.STATE_ROOM_POWER_LEVELS, QueryStringValue.NoCondition)
+                        .mapOptional { it.content.toModel<PowerLevelsContent>() }
+                        .unwrap()
+        ) { roomMembers, powerLevelsContent ->
+            buildAdminMembersList(powerLevelsContent, roomMembers)
+        }
+                .execute { async ->
+                    copy(isLastAdmin = async.invoke()?.size == 1 &&
+                            async.invoke()?.find { roomMember ->
+                                roomMember.userId == session.myUserId
+                            } != null)
+                }
```

Comment:
Not sure about the difference between execute and onEach but I suggest you these changes.
- `SingleOrNull` does the trick to check if there is a single element and return it if any.
- `OnEach` will be triggered once Flows are updated and do not return an `async` with a nullable value when invoking it.

```suggestion
                .distinctUntilChanged()
                .onEach { adminsMembers ->
                    setState {
                        copy(isLastAdmin = adminsMembers.singleOrNull()?.userId == session.myUserId)
                    }
                }
                .launchIn(viewModelScope)
```